mixer/cmd/build.go

```diff
@@ -199,113 +213,163 @@ var buildFormatNewCmd = &cobra.Command{
+			}
+			return nil
+		}
+		fmt.Println("Removing deprecated bundle-info files...")
+		// Remove deleted bundles and replace with empty dirs for update to mark as deleted
+		if err = b.ModifyBundles(f); err != nil {
 			fail(err)
 		}
-		// Fill this in w/Update bundle definitions
-		// if err := UpdateBudlesForFormatBump(); err != nil {...}
 
-		// Build the +20 (first build in new format) bundles
-		if err = buildBundles(b, buildFlags.noSigning, buildFlags.clean); err != nil {
+		// Link the +20 bundles to the +10 so we are building the update with the same
+		// underlying content. The only things that might change are the manifests
+		// (potentially the pack and full-file formats as well, though this is very
+		// rare).
+		ver, err = strconv.Atoi(b.MixVer)
+		if err != nil {
 			fail(err)
 		}
+		ver -= 10
+		source := filepath.Join(b.Config.Builder.ServerStateDir, "image", b.MixVer)
+		dest := filepath.Join(b.Config.Builder.ServerStateDir, "image", strconv.Itoa(ver))
+		fmt.Println(" Copying +20 bundles to +10 bundles")
+		if err = helpers.RunCommandSilent("cp", "-al", source, dest); err != nil {
+			failf("Failed to copy +10 bundles to +20: %s\n", err)
+		}
+
+		// Set the format back to old for the actual build update
+		b.State.Mix.Format = oldFormat
 
+		// Update mixer to build version +10, the version we want to output for old format
 		ver, err = strconv.Atoi(b.MixVer)
 		if err != nil {
 			fail(err)
 		}
-
-		// Build the +20 update so we don't have to switch tooling in between
+		if err = b.UpdateMixVer(ver - 10); err != nil {
+			failf("Couldn't update Mix Version")
+		}
+		// Build the update content for the +10 build
 		params := builder.UpdateParameters{
-			MinVersion:    ver,
-			Format:        strconv.Itoa(newFormat),
+			MinVersion:    buildFlags.minVersion,
+			Format:        b.State.Mix.Format,
 			Publish:       !buildFlags.noPublish,
 			SkipSigning:   buildFlags.noSigning,
 			SkipFullfiles: buildFlags.skipFullfiles,
 			SkipPacks:     buildFlags.skipPacks,
 		}
-		err = b.BuildUpdate(params)
-		if err != nil {
+		if err = b.BuildUpdate(params); err != nil {
 			failf("Couldn't build update: %s", err)
 		}
 
-		// Copy +20 chroots to +10 so we can build last formatN build with the
-		// same content
-		prevVersion, err := strconv.Atoi(b.MixVer)
-		if err != nil {
-			fail(err)
-		}
-		prevVersion -= 10
-		source := filepath.Join(b.Config.Builder.ServerStateDir, "image", b.MixVer)
-		dest := filepath.Join(b.Config.Builder.ServerStateDir, "image", strconv.Itoa(prevVersion))
-		fmt.Println(" Copying +20 chroots to +10 chroots")
-		if err = helpers.RunCommandSilent("cp", "-al", source, dest); err != nil {
-			failf("Failed to copy +20 chroots to +10: %s\n", err)
+		// Write the +0 version to LAST_VER so that we reference in both +10 and +20 manifests as the 'previous:'
+		if err = ioutil.WriteFile(filepath.Join(b.Config.Builder.ServerStateDir, "image", "LAST_VER"), []byte(strconv.Itoa(ver-20)), 0644); err != nil {
```

Comment:
wait `ver-20`? isn't `ver` already at the +10 version, so this would put it at the -10 version?

ohhhh no you reread `ver` from `b.MixVer` and just did `ver - 10` not `ver -= 10`. This is confusing lol.